### PR TITLE
Fix simulate remove ingest processor throwing illegal_argument_exception

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
This PR fixes the bug of simulating remove ingest processor always throwing illegal_argument_exception, the bug is caused by the new merged [PR](https://github.com/opensearch-project/OpenSearch/pull/10895) which disallowing removing some metadata fields, but when simulating processors, the ingesting document does not have the metadata fields like `_version` and `_version_type`, so `document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class)` throws illegal_argument_exception rather than return null value, in order to return null value, use another method instead with setting `ignore_missing` to `true`.

Both the unit test code and yml test do not cover this case, I've added some yml tests and unit test code for it.

### Related Issues
https://github.com/opensearch-project/OpenSearch/issues/11604

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Failing checks are inspected and point to the corresponding known issue(s) (See: [Troubleshooting Failing Builds](../blob/main/CONTRIBUTING.md#troubleshooting-failing-builds))
- [x] Commits are signed per the DCO using --signoff
- [x] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))
- [x] Public documentation issue/PR [created](https://github.com/opensearch-project/documentation-website/issues/new/choose)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gaobinlong** [comment]
@reta , could you help to review this PR? Thanks!

File: modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java [hunk 1]
```diff
@@ -88,16 +88,18 @@ public IngestDocument execute(IngestDocument document) {
...
-                Long version = document.getFieldValue(IngestDocument.Metadata.VERSION.getFieldName(), Long.class);
-                throw new IllegalArgumentException(
-                    "cannot remove metadata field [_id] when specifying external version for the document, version: "
-                        + version
-                        + ", version_type: "
-                        + versionType
-                );
+            if (path.equals(IngestDocument.Metadata.ID.getFieldName())) {
+                String versionType = document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class, true);
+                if (Objects.equals(versionType, VersionType.toString(VersionType.EXTERNAL))
```
Comment (reta) [inline modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java:93]: The previously implemented check was more concise and compact:
 
```suggestion
                if (!Objects.equals(versionType, VersionType.toString(VersionType.INTERNAL))
```

File: modules/ingest-common/src/test/java/org/opensearch/ingest/common/RemoveProcessorTests.java [hunk 2]
```diff
@@ -180,5 +180,21 @@ public void testRemoveMetadataField() throws Exception {
                         assertThat(ingestDocument.hasField(metadataFieldName), equalTo(false));
                     }
         }
+
+        // test remove _id when _version_type is null
```
Comment (reta) [inline modules/ingest-common/src/test/java/org/opensearch/ingest/common/RemoveProcessorTests.java:184]: Could you please extract it into separate test method (if possible)?

File: modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java (hunk 1 above)
Comment (gaobinlong) [reply modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java:93]: We have to check if `versionType` is null, if it's null, no exception will be thrown, something like this:
```
if (versionType!=null && !Objects.equals(versionType, VersionType.toString(VersionType.INTERNAL)){
...
}
```
, do you still think this is clearer than the positive logic?

Comment (reta) [reply modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java:93]: Ah I see you changed `document.getFieldValue` to use "ignore missing" favor, may be we could do:

```
if (path.equals(IngestDocument.Metadata.ID.getFieldName()) && document.hasField(IngestDocument.Metadata.VERSION_TYPE.getFieldName()) {
```

Comment (gaobinlong) [reply modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java]: Thanks, that makes sense, I've changed the code.

File: modules/ingest-common/src/test/java/org/opensearch/ingest/common/RemoveProcessorTests.java (hunk 2 above)
Comment (gaobinlong) [reply modules/ingest-common/src/test/java/org/opensearch/ingest/common/RemoveProcessorTests.java:262]: I've added a new test method for it.

**Review by reta — APPROVED** [review]

**Comment by gaobinlong** [comment]
Hi @reta , can this PR be merged now? I've resolved the conflict and now all checks have passed yet.

**Review by reta — APPROVED** [review]

**Pull request merged**